pyquil/wavefunction.py

```diff
@@ -145,28 +146,43 @@ def pretty_print_probabilities(self, decimal_digits=2):
                 outcome_dict[outcome] = prob
         return outcome_dict
 
-    def pretty_print(self, decimal_digits=2):
+    def pretty_print(self, decimal_digits=2, ancillas: Optional[List[int]] = None):
         """
         Returns a string repr of the wavefunction, ignoring all outcomes with approximately zero
         amplitude (up to a certain number of decimal digits) and rounding the amplitudes to
         decimal_digits.
 
         :param int decimal_digits: The number of digits to truncate to.
+        :param List[int] ancillas: The list of indices for ancillary qubits to omit.
         :return: A dict with outcomes as keys and complex amplitudes as values.
         :rtype: str
         """
+
+        if ancillas is None:
+            ancillas = []
+
+        mask = 0
+        for ancilla in ancillas:
+            mask |= (1 << ancilla)
+
         outcome_dict = {}
         qubit_num = len(self)
         pp_string = ""
         for index, amplitude in enumerate(self.amplitudes):
+            if index & mask != 0 and not np.isclose(amplitude, 0):
+                raise ValueError('Specified ancilla was found to have non-zero amplitude.')
```

Comment:
just for my own edification -- is this a strictly necessary requirement for this feature?

Reply:
This is what I came up with from a conversation with @ecpeterson, but now that you mention it, I'm not sure it's necessary to check for non-zero amplitudes. Thoughts?